### PR TITLE
Fix errors in Weapon::Get_Percent_Ready_To_Fire

### LINKED ISSUE #916: Assert in Weapon::Get_Percent_Ready_To_Fire() is hit after using clear mines ability
Assert is hit after using clear mines ability:

```
captainslog_dbgassert(time_left <= total_time, "time_left <= total_time");
```

```
>	thyme.dll!Weapon::Get_Percent_Ready_To_Fire() Line 1026	C++
 	game.dat!005a45c4()	Unknown
 	game.dat![Frames below may be incorrect and/or missing, no symbols loaded for game.dat]	Unknown
 	game.dat!005a3cf9()	Unknown
 	game.dat!0045e9e8()	Unknown
 	game.dat!0050a6fa()	Unknown
 	game.dat!004ad301()	Unknown
 	thyme.dll!GameEngine::Update() Line 176	C++
 	thyme.dll!Win32GameEngine::Update() Line 58	C++
 	thyme.dll!GameEngine::Execute() Line 556	C++
 	thyme.dll!Game_Main(int argc, char * * argv) Line 41	C++
 	thyme.dll!main(int argc, char * * argv) Line 700	C++
 	thyme.dll!Main_Func(HINSTANCE__ * hInstance, HINSTANCE__ * hPrevInstance, char * lpCmdLine, int nCmdShow) Line 728	C++
```

### PR DESCRIPTION
**Pull request by bobtista**
* Fixes #916

### CODE REVIEW

File: src/game/logic/object/weapon.cpp [hunk 1]
```diff
@@ -1019,10 +1019,10 @@ float Weapon::Get_Percent_Ready_To_Fire() const
 
             if (now < next_shot) {
                 captainslog_dbgassert(next_shot >= m_whenLastReloadStarted, "next_shot >= m_whenLastReloadStarted");
-                unsigned int time_left = next_shot - m_whenLastReloadStarted;
```
Comment (bobtista) [inline src/game/logic/object/weapon.cpp:1022]: Should we also replace m_whenLastReloadStarted with the getter Get_Last_Reload_Started()?

**Comment by jonwil** [comment]
Having reviewed the original code, the correct fix is to rename time_left to total_time and total_time to time_left.
So
unsigned int time_left = next_shot - m_whenLastReloadStarted; 
becomes
unsigned int total_time = next_shot - m_whenLastReloadStarted;
and
unsigned int total_time = next_shot - m_whenLastReloadStarted;
becomes
unsigned int time_left = next_shot - m_whenLastReloadStarted;

and the other uses of both variables get swapped as well.

And no we should not use the getter in this case.

File: src/game/logic/object/weapon.cpp (hunk 1 above)
Comment (tomsons26) [reply src/game/logic/object/weapon.cpp:1022]: original code didn't and seems they followed a rule to avoid getters inside classes that define the member

**Comment by bobtista** [comment]
@jonwil do you mean that total_time should be defined first? Like
`unsigned int total_time = next_shot - m_whenLastReloadStarted;

                if (time_left != 0) {
                    unsigned int time_left = next_shot - now;`
If so, I should change the check to if (total_time != 0)
Or, I could both rename and swap the order of the definitions such that the check if (time_left != 0) stays the same

**Comment by xezon** [comment]
Does change fix #916 confirmed?

Does it perhaps also fix #888?

**Review by jonwil — APPROVED** [review]
Verified that the change in this PR is correct. Approved.

**Comment by xezon** [comment]
I will check later if this fixes the bugs.

**Comment by xezon** [comment]
* Fixes #916
* Does not fix #888

**Pull request merged**